modules/core/src/test/java/org/apache/ignite/util/GridCommandHandlerTest.java

```diff
@@ -2599,29 +2602,85 @@ public void testDiagnosticPageLocksTracker() throws Exception {
-        assertEquals(EXIT_CODE_OK, execute("--diagnostic", "pageLocks", "dump_log"));
-        assertEquals(EXIT_CODE_OK, execute("--diagnostic", "pageLocks", "dump", dir));
+        if (testDir.exists())
+            assertTrue(U.delete(testDir));
 
-        assertEquals(EXIT_CODE_OK, execute("--diagnostic", "pageLocks", "dump", "--all"));
-        assertEquals(EXIT_CODE_OK, execute("--diagnostic", "pageLocks", "dump_log", "--all"));
-        assertEquals(EXIT_CODE_OK, execute("--diagnostic", "pageLocks", "dump", dir, "--all"));
+        boolean success = false;
 
-        assertEquals(EXIT_CODE_OK, execute("--diagnostic", "pageLocks", "dump", "--nodes",
-            node0.id().toString(), node2.id().toString()));
-        assertEquals(EXIT_CODE_OK, execute("--diagnostic", "pageLocks", "dump", "--nodes",
-            node0.consistentId().toString(), node2.consistentId().toString()));
+        try {
+            assertEquals(EXIT_CODE_OK, execute("--diagnostic"));
+            assertEquals(EXIT_CODE_OK, execute("--diagnostic", "help"));
+            assertEquals(EXIT_CODE_OK, execute("--diagnostic", "pageLocks", "help"));
+
+            assertEquals(EXIT_CODE_OK, execute("--diagnostic", "pageLocks", "dump"));
+
+            checkNumberFiles(default0, 1);
+
+            assertEquals(EXIT_CODE_OK, execute("--diagnostic", "pageLocks", "dump_log"));
+            assertEquals(EXIT_CODE_OK, execute("--diagnostic", "pageLocks", "dump", "--path",
+                testDir.getAbsolutePath()));
+
+            checkNumberFiles(testDir, 1);
+
+            assertEquals(EXIT_CODE_OK, execute("--diagnostic", "pageLocks", "dump", "--all"));
+
+            checkNumberFiles(default0, 5);
+
+            assertEquals(EXIT_CODE_OK, execute("--diagnostic", "pageLocks", "dump_log", "--all"));
+            assertEquals(EXIT_CODE_OK, execute("--diagnostic", "pageLocks", "dump",
+                "--path", testDir.getAbsolutePath(), "--all"));
+
+            checkNumberFiles(testDir, 5);
+
+            assertEquals(EXIT_CODE_OK, execute("--diagnostic", "pageLocks", "dump",
+                "--nodes", node0.id().toString() + "," + node2.id().toString()));
+
+            checkNumberFiles(default0, 7);
+
+            assertEquals(EXIT_CODE_OK, execute("--diagnostic", "pageLocks", "dump",
+                "--nodes", node0.consistentId().toString() + "," + node2.consistentId().toString()));
+
+            assertEquals(EXIT_CODE_OK, execute("--diagnostic", "pageLocks", "dump_log",
+                "--nodes", node1.id().toString() + "," + node3.id().toString()));
+
+            assertEquals(EXIT_CODE_OK, execute("--diagnostic", "pageLocks", "dump",
+                "--path", testDir.getAbsolutePath(),
+                "--nodes", node1.consistentId().toString() + "," + node3.consistentId().toString()));
+
+            checkNumberFiles(testDir, 7);
+
+            success = true;
+        }
+        finally {
```

Comment:
You should move cleaning files/directories from test to `befoteTest()`/`afterTest()` methods